File: docs/sphinx/snippets/python/default_ops/h_op.py
import cudaq

def main():
    qubit = cudaq.qubit()
    # [Begin H Op]
    # Apply the unitary transformation
    # H = (1 / `sqrt`(2)) * | 1   1 |
    #                     | 1  -1 |
    h(qubit)
    # [End H Op]
if __name__ == "__main__":
    main()

File: docs/sphinx/snippets/cpp/default_ops/h_op.cpp
/*******************************************************************************
 * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// Compile and run with: `nvq++ h_op.cpp && ./a.out`

#include <cudaq.h>

int main() {
  cudaq::qubit qubit;

  // Apply the unitary transformation
  // H = (1 / sqrt(2)) * | 1   1 |
  //                     | 1  -1 |
  h(qubit);

  return 0;
}

